services/order.py

```diff
@@ -0,0 +1,44 @@
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, User
+from datetime import datetime
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: datetime = None
+) -> Order:
+    with transaction.atomic():
```

Comment:
You could use `transaction.atomic()` as a decorator instead of a context manager in case all function body should be treated as an atomic transaction.